frontend/packages/ceph-storage-plugin/src/components/ocs-install/ocs-install.tsx

```diff
@@ -0,0 +1,365 @@
+            defaultRequestSizeValue={requestSizeValue}
+            dropdownUnits={dropdownUnits}
+          />
+        </div>
+        <div className="form-group">
+          <Tooltip content="The Storage Class will be used to request storage from the underlying infrastructure to create the backing persistent volumes that will be used to provide the OpenShift Container Storage (OCS) service">
+            <OCSStorageClassDropdown
+              onChange={handleStorageClass}
+              id="storageclass-dropdown"
+              required={false}
+              name="storageClass"
+            />
+          </Tooltip>
+        </div>
+        <ButtonBar errorMessage={error} inProgress={inProgress}>
+          <ActionGroup className="pf-c-form">
+            <Button type="submit" variant="primary" isDisabled={createBtnDisabled}>
+              Create
+            </Button>
+            <Button type="button" variant="secondary" onClick={history.goBack}>
+              Cancel
+            </Button>
+          </ActionGroup>
+        </ButtonBar>
+      </form>
+    </div>
+  );
+});
+
+export const CreateOCSServiceYAML: React.FC<CreateOCSServiceYAMLProps> = (props) => {
+  const template = _.attempt(() => safeDump(props.sample));
+  if (_.isError(template)) {
+    // eslint-disable-next-line no-console
+    console.error('Error parsing example JSON from annotation. Falling back to default.');
+  }
+
+  return (
+    <CreateYAML
+      {...props as any}
+      template={template}
+      match={props.match}
+      hideHeader
+      plural={OCSServiceModel.plural}
+    />
+  );
+};
+
+/**
+ * Component which wraps the YAML editor and form together
+ */
+export const CreateOCSService: React.FC<CreateOCSServiceProps> = React.memo((props) => {
+  const [sample, setSample] = React.useState(null);
+  const [method, setMethod] = React.useState<'yaml' | 'form'>('form');
+  const [clusterServiceVersion, setClusterServiceVersion] = React.useState(null);
+
+  React.useEffect(() => {
+    k8sGet(ClusterServiceVersionModel, props.match.params.appName, props.match.params.ns)
+      .then((clusterServiceVersionObj) => {
+        try {
+          setSample(
+            JSON.parse(_.get(clusterServiceVersionObj.metadata.annotations, 'alm-examples'))[0],
+          );
+          setClusterServiceVersion(clusterServiceVersionObj);
+        } catch (e) {
+          setClusterServiceVersion(null);
+        }
+      })
+      .catch(() => setClusterServiceVersion(null));
+  }, [props.match.params.appName, props.match.params.ns]);
+
+  return (
+    <React.Fragment>
+      <div className="co-create-operand__header">
+        <div className="co-create-operand__header-buttons">
+          {clusterServiceVersion !== null && (
+            <BreadCrumbs
+              breadcrumbs={[
+                {
+                  name: clusterServiceVersion.spec.displayName,
+                  path: window.location.pathname.replace('/~new', ''),
```

Comment:
I'd rather not use path math here. Is there a helper for generating the path from the resource?